src/examples/chapter-10/README.md

```diff
@@ -0,0 +1,154 @@
+
+## Features
+
+- **PDF Document Processing**: Extracts and processes text from PDF files
+- **Vector Embeddings**: Uses OpenAI's text-embedding-3-large model for semantic embeddings
+- **Local Vector Storage**: Stores embeddings locally using localvec plugin
+- **Semantic Search**: Retrieves relevant document chunks based on user queries
+- **RAG Implementation**: Combines retrieved context with LLM responses
+- **Two-Phase Architecture**: Separate indexing and retrieval flows
+
+## System Components
+
+### 1. Indexer Flow
+Processes PDF documents and creates searchable vector embeddings:
+- Reads PDF files (default: Arduino report)
+- Extracts and chunks text content
+- Generates embeddings using OpenAI
+- Stores vectors in local database
+
+### 2. Retrieval Flow
+Performs semantic search and generates contextualized responses:
+- Takes user queries as input
+- Searches vector database for relevant content
+- Retrieves top-K most similar documents
+- Generates AI responses enhanced with retrieved context
+
+## AI Flows
+
+### 1. Indexer Flow
+**Endpoint:** `POST /indexerFlow`
+
+Indexes documents for search. Use this to process and store PDF content.
+
+**Request Body:**
+```json
+{
+  "pdfPath": "path/to/document.pdf"  // Optional, defaults to Arduino report
+}
+```
+
+**Example:**
+```bash
+curl -X POST http://localhost:9090/indexerFlow \
+  -H "Content-Type: application/json" \
+  -d '{"pdfPath": "internal/docs/arduino_report.pdf"}'
+```
+
+### 2. Retrieval Flow
+**Endpoint:** `POST /retrievalFlow`
+
+Searches indexed documents and generates contextual responses.
+
+**Request Body:**
+```json
+{
+  "query": "Your question here",
+  "k": 5  // Optional, number of results to retrieve (default: 5)
+}
+```
+
+**Example:**
+```bash
+curl -X POST http://localhost:9090/retrievalFlow \
+  -H "Content-Type: application/json" \
+  -d '{"query": "What are the key features of Arduino?", "k": 3}'
+```
+
+## Setup and Running
+
+### Prerequisites
+- Go 1.24 or later
+- OpenAI API key
+
+### 1. Environment Setup
+Set your OpenAI API key:
+```bash
+export OPENAI_API_KEY="your-openai-api-key-here"
+```
+
+Optionally set a custom port (defaults to 9090):
+```bash
+export PORT="3000"
+```
+
+### 2. Install Dependencies
+```bash
+go mod download
+```
+
+### 3. Run the Application
+```bash
+go run main.go
+```
+
+## Usage Workflow
+
+### Step 1: Index Documents
+First, run the indexer to process and store your documents:
+```bash
+curl -X POST http://localhost:9090/indexerFlow \
+  -H "Content-Type: application/json" \
+  -d '{}'
+```
+
+### Step 2: Query the System
+Once indexing is complete, you can query the system:
+```bash
+curl -X POST http://localhost:9090/retrievalFlow \
+  -H "Content-Type: application/json" \
+  -d '{"query": "How does Arduino work?"}'
```

Comment:
Genkit HTTP handlers expect the request body to be wrapped in a `data` field.
`  -d '{"data":{"query": "How does Arduino work?"}}'`

Reply:
fixed